src/Microsoft.PowerShell.Commands.Utility/commands/utility/join-string.cs

```diff
@@ -0,0 +1,210 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Management.Automation.Internal;
+using System.Management.Automation.Language;
+using System.Text;
+
+namespace Microsoft.PowerShell.Commands.Utility
+{
+    /// <summary>
+    /// Join-Object implementation.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Join, "String", RemotingCapability = RemotingCapability.None, DefaultParameterSetName = "default")]
+    [OutputType(typeof(string))]
+    public sealed class JoinStringCommand : PSCmdlet
+    {
+        private const int Capacity = 50;
+
+        // ReSharper disable once CollectionNeverQueried.Local
+        private readonly List<PSObject> _inputObjects = new List<PSObject>(Capacity);
+        private DynamicPropertyGetter _propGetter = new DynamicPropertyGetter();
+
+        /// <summary>
+        /// Gets or sets the property name or script block to use as the value to join.
+        /// </summary>
+        [Parameter(Position = 0)]
+        [ArgumentCompleter(typeof(PropertyNameCompleter))]
+        public object Property { get; set; }
+
+        /// <summary>
+        /// Gets or sets the delimiter to join the output with.
+        /// </summary>
+        [Parameter(Position = 1)]
+        [ArgumentCompleter(typeof(JoinItemCompleter))]
+        [AllowEmptyString]
+        public string Delimiter { get; set; }
+
+        /// <summary>
+        /// Gets or sets text to include before the joined input text.
+        /// </summary>
+        [Parameter]
+        public string Prefix { get; set; }
+
+        /// <summary>
+        /// Gets or sets text to include after the joined input text.
+        /// </summary>
+        [Parameter]
+        public string Suffix { get; set; }
+
+        /// <summary>
+        /// Gets or sets if the output items should we wrapped in single quotes.
+        /// </summary>
+        [Parameter(ParameterSetName = "Quote")]
+        public SwitchParameter Quote { get; set; }
```

Comment:
SingleQuote supplements DoubleQuote.

Another thought is to make one parameter (which replace both Quote and DoubleQuote parameters) with allowed values SingleQuote and DoubleQuote. Seems better UX. This also reduce account of parameter sets.

Reply:
I don't. I have used this cmdlet in a slightly different but very similar implementation, and the most common use case, by far, is to create strings for use in code.
Having `DoubleQuote` and `SingleQuote` is so much nicer than to have to specify a `Format`. 

I can for sure see the value of Format, but it is not as nice from an interactive perspective as the switches.